### PR TITLE
Add FXIOS-10667 Content blocklist support via Remote Settings

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10667)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/23336)

## :bulb: Description

Initial updates to support Remote Settings for content blocking lists in iOS client. Changes:
- Removes our previous Shavar shell script and related code in `bootstrap.sh`
- Fixes old Xcode project references to `disconnect-*` files
- Updates older code that was loading the JSON data directly from bundle to instead route that through the Remote Settings APIs so we have more control over how that data is vended

This is still a work-in-progress and there are a few areas that will need to be updated further (I have not added or updated tests etc.).

### Result

```
2024-11-22 15:52:23.738 💙 INFO [adblock] ContentBlocker - Compiling any lists not already in rule store...
2024-11-22 15:52:23.738 💙 INFO [adblock] ContentBlocker - Will compile list: disconnect-block-advertising
2024-11-22 15:52:23.745 💙 INFO [adblock] ContentBlocker - Will compile list: disconnect-block-analytics
2024-11-22 15:52:23.745 💙 INFO [adblock] ContentBlocker - Will compile list: disconnect-block-social
2024-11-22 15:52:23.745 💙 INFO [adblock] ContentBlocker - Will compile list: disconnect-block-cryptomining
2024-11-22 15:52:23.746 💙 INFO [adblock] ContentBlocker - Will compile list: disconnect-block-fingerprinting
2024-11-22 15:52:23.746 💙 INFO [adblock] ContentBlocker - Will compile list: disconnect-block-cookies-advertising
2024-11-22 15:52:23.746 💙 INFO [adblock] ContentBlocker - Will compile list: disconnect-block-cookies-analytics
2024-11-22 15:52:23.746 💙 INFO [adblock] ContentBlocker - Will compile list: disconnect-block-cookies-social
2024-11-22 15:52:24.147 💙 INFO [adblock] ContentBlocker - Compiled 8 of 8 lists checked. 0 errors.
```

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Review by nbhasin2 — APPROVED** [review]

File: bootstrap.sh
Comment (mattreaganmozilla) [inline bootstrap.sh:51]: cc @clarmso Just a quick heads up that we are removing this content blocker update script from the bootstrap.sh. 

I'm not aware of any issues this should cause on Bitrise but wanted to make sure you were looped in. Thanks

Comment (mattreaganmozilla) [reply bootstrap.sh:51]: I just noticed that this script is also responsible for running the code that generates our `MainFrameAtDocument*.js` files. So I'm fixing this right now, should have an update pushed soon.

**Comment by mattreaganmozilla** [comment]
I believe the way the script is removed currently will break the generation of our `MainFrameAtDocument*.js` files, adding Do Not Merge until a fix is pushed up for that. Taking a closer look.

File: bootstrap.sh
Comment (mattreaganmozilla) [reply bootstrap.sh:51]: (Update: the above issue should be fixed now.)

File: firefox-ios/Client/ContentBlocker/ContentBlocker.swift
Comment (issammani) [inline firefox-ios/Client/ContentBlocker/ContentBlocker.swift:310]: Not really something we can support right now, but once we figure out caching we can also get the file digest from RS for attachments.

File: firefox-ios/Client/Application/RemoteSettings/RemoteDataType.swift
Comment (issammani) [inline firefox-ios/Client/Application/RemoteSettings/RemoteDataType.swift:54]: Eventually I think we would want to converge to the desktop implementation where a `record` and `attachment` is a one to one mapping. So the API might look like `record.get_attachment`. But this also works.

**Review by issammani — APPROVED** [review]
This looks @mattreaganmozilla. Thanks. I see some stuff we can improve in the future, but this a really good place to start.

File: bootstrap.sh
Comment (issammani) [inline bootstrap.sh:52]: Nice catch. Not sure why npm was part of `content_blocker_update.sh` before 🤔

File: content_blocker_update.sh
Comment (mattreaganmozilla) [inline content_blocker_update.sh:20]: Note: I believe removal of this also means we can clean up some of the related code in BrowserKit (e.g. `ContentBlockerGenerator`)?

Comment (issammani) [reply content_blocker_update.sh:20]: Yes. I converted it to a simpler python script. My initial plan was to use it as is and have docker multistage build ( like I did in [this tmp repo](https://github.com/issammani/tracking-protection-lists-webkit/blob/7510086b054a3129a5aebbf708047a1acaa571b0/Dockerfile#L27-L38 ) ) to build a binary from swift and use it, but that overcomplicates the build process. We don't need that code anymore.

Comment (mattreaganmozilla) [reply content_blocker_update.sh:20]: Ok thanks, I might create a follow-up task for that cleanup and just put up a quick separate PR to remove it.

File: firefox-ios/Client/Application/RemoteSettings/RemoteDataType.swift
Comment (nbhasin2) [reply firefox-ios/Client/Application/RemoteSettings/RemoteDataType.swift:54]: @issammani Once this lands, I can add a patch for you two to review the attachment portion where yes you can do some part in python however locally it should all work out in terms of record.getAttachment(...name)

**Pull request merged**
